libvast/src/table_slice.cpp

```diff
@@ -113,22 +90,56 @@ void table_slice::append_column_to_index(size_type col,
     idx.append(at(row, col), offset() + row);
 }
 
-table_slice_ptr make_table_slice(record_type layout, caf::actor_system& sys,
-                                 caf::atom_value impl,
-                                 table_slice::size_type rows) {
-  if (impl == default_table_slice::class_id) {
-    return caf::make_copy_on_write<default_table_slice>(std::move(layout));
+bool add_table_slice_factory(caf::atom_value id, table_slice_factory f) {
+  if (factory_.count(id) > 0)
```

Comment:
In the unit tests, the fixture may try to register a factory multiple times, so we need to perform the check. Since we're already checking, I think we can keep the `bool` return value. In application code, it's probably discarded anyway.